### DISCUSSION TITLE
Starting an action manually via 'Run workflow' will erase inputs after switching branches

### DISCUSSION THREAD

**Discussion by amleszk** [community/community#15625] [Actions]
When starting a workflow from Github actions 'Run workflow' UI, if you enter parameters into the form and then switch the branch, the parameters that got entered will disappear or revert back to defaults. Can you fix it so switching branches does not change the entered parameters?

In example screenshot I would enter parameter for the form entry text boxes, then forget I need to switch to another branch other than 'master'. When switching branches the parameters are removed or replaced with defaults.

<img width="308" alt="image" src="https://user-images.githubusercontent.com/501631/165295089-ad2ab90d-12ee-44d3-a2bb-98d6aa24df2a.png">

**Comment by philomory** [comment]
I'm not sure that would really work in the general case, since the same workflow in different branches can have different inputs defined. Though, even if it simply retained the entered values when the workflow definition on the newly-selected branch has identical inputs to those from the previously-selected branch, that would probably be good enough.

**Reply by amleszk** [reply]
Yep if the parameter definition changed i would expect the value to get removed, but mostly the parameter definitions are the same
